### ISSUE TITLE
Create typescript definitions

### ISSUE THREAD

**Issue by dgileadi** [moll/js-must#37]
Some tools (like [Visual Studio Code](https://code.visualstudio.com)) can use typescript definitions for code insight and completion. Many popular JS libraries [already have typescript definitions](http://definitelytyped.org). It would be really nice to have typescript definitions for js-must too.

In any case thanks for a great library!

**Comment by moll** [comment]
:+1:

I was thinking of some just today while giving Flow a try. Perhaps it's possible to autogenerate both from inline documentation...

**Comment by moll** [comment]
@dgileadi: @KarlPurk added some type definitions. Would you mind giving them a spin, too?

**Comment by moll** [comment]
Let's carry on from https://github.com/moll/js-must/pull/48 here, @KarlPurk .

:+1: for the tests, too. What do you have in mind? I can imagine positive cases are easy to test — everything in some file has to pass. How would you do negative ones? "This file has to have this number of failures"? Or maybe negative cases here are too much work for little benefit.

**Comment by KarlPurk** [comment]
Yes I think there's little benefit to negative testing, the main value in testing the definition file is to make sure it covers the API fully (and correctly) so we likely just need to test the happy path.
